clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6158,12 +6158,24 @@ def err_illegal_initializer_type : Error<"illegal initializer type %0">;
 def ext_init_list_type_narrowing : ExtWarn<
   "type %0 cannot be narrowed to %1 in initializer list">,
   InGroup<CXX11Narrowing>, DefaultError, SFINAEFailure;
+// *_narrowing_const_reference diagnostics have the same messages, but are
+// controlled by -Wc++11-narrowing-const-reference for narrowing involving a
+// const reference.
```

Comment:
FYI, you can give them the same text without duplicating the text by doing something like:

```
def warn_objc_pointer_masking_performSelector : Warning<warn_objc_pointer_masking.Summary>,
  InGroup<ObjCPointerIntrospectPerformSelector>;
```

Reply:
@nico: if you haven't already, please do this in a follow up patch (since you've merged this!).

Reply:
I created #77278 to simplify the code.